bigquery/google/cloud/bigquery/client.py

```diff
@@ -525,6 +544,171 @@ def load_table_from_storage(self, source_uris, destination,
         job.begin()
         return job
 
+    def load_table_from_file(self, file_obj, destination,
+                             rewind=False,
+                             size=None,
+                             num_retries=_DEFAULT_NUM_RETRIES,
+                             job_id=None, job_config=None):
+        """Upload the contents of this table from a file-like object.
+
+        Like load_table_from_storage, this creates, starts and returns
+        a ``LoadJob``.
+
+        :type file_obj: file
+        :param file_obj: A file handle opened in binary mode for reading.
+
+        :type destination: :class:`google.cloud.bigquery.table.TableReference`
+        :param destination: Table into which data is to be loaded.
+
+        :type rewind: bool
+        :param rewind: If True, seek to the beginning of the file handle before
+                       writing the file.
+
+        :type size: int
+        :param size: The number of bytes to read from the file handle.
+                     If not provided, we'll try to guess the size using
+                     :func:`os.fstat`. (If the file handle is not from the
+                     filesystem this won't be possible.)
+
+        :type num_retries: int
+        :param num_retries: Number of upload retries. Defaults to 6.
+        :type job_id: str
+        :param job_id: (Optional) Name of the job.
+
+        :type job_config: :class:`google.cloud.bigquery.job.LoadJobConfig`
+        :param job_config: (Optional) Extra configuration options for the job.
+
+        :rtype: :class:`~google.cloud.bigquery.jobs.LoadJob`
+
+        :returns: the job instance used to load the data (e.g., for
+                  querying status). Note that the job is already started:
+                  do not call ``job.begin()``.
+        :raises: :class:`ValueError` if ``size`` is not passed in and can not
+                 be determined, or if the ``file_obj`` can be detected to be
+                 a file opened in text mode.
+        """
+        job_id = _make_job_id(job_id)
+        job = LoadJob(job_id, None, destination, self, job_config)
+        job_resource = job._build_resource()
+        if rewind:
+            file_obj.seek(0, os.SEEK_SET)
+        _check_mode(file_obj)
+        try:
+            if size is None:
```

Comment:
I find this a bit odd to switch between resumable upload and multipart upload based solely on the size being none. We should also check if the size is not none but too big.

Based on what I know about other Apiary endpoints [multipart upload should only be used for small files](https://cloud.google.com/storage/docs/json_api/v1/how-tos/multipart-upload).

> For larger files (more than 5 MB) or less reliable network connections, use resumable upload instead.

IIRC, we have got feedback from users that uploads with can fail. I think this could be the reason.

Reply:
Done.